app/js/enhancedDate.js

```diff
@@ -1,5 +1,80 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function iife() {
+    
+    // private functions
+    var publicAPI = {
+        setDate: function setDate(userDate) {
+            // Set working date to user input if date object given
+            if (userDate instanceof Date) {
+                var workingDate = userDate;
+            } else {
+                var workingDate = new Date();
+            }
 
+            return workingDate;
+        },
+
+        getDate: function getDate(asDateObj) {
+            // return date object if requested
+            if (asDateObj === true) {
+                return workingDate;
+            } else {
+            // return milliseconds after 1970 by default
+                
+                return workingDate.getTime();
+            }
+        },
+
+        getDayName: function getDayName() {
+            // return weekday name string
+
+            // define lookup for weekday value to name
+            var weekday = { 0: 'Sunday',
```

Comment:
This is the right way to do this, but it would be better to put this inside your module but outside of the method. As it stands, it will re-create this object every time this function is called.